### PR TITLE
test: improve and refactor SE attester test coverage

### PR DESCRIPTION
**Pull request by SantoshMadhukar-K**
Updated the test suite using rstest parametric suites for UserData, validate_and_pad_data, and get_evidence error paths, and added new test coverage for SeAttestationResponse serde, Debug output, and edge cases not previously tested.

```
cargo test -p attester --no-default-features --features se-attester --lib se   
   Compiling attester v0.1.0 (/root/temp_repos/myforks/guest-components/attestation-agent/attester)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 8m 21s
     Running unittests src/lib.rs (/root/temp_repos/myforks/guest-components/target/debug/deps/attester-fbab0354e7141812)

running 32 tests
test se::tests::test_get_evidence_early_failure::case_1_invalid_json ... ok
test se::tests::test_get_evidence_early_failure::case_2_empty_input ... ok
test se::tests::test_get_evidence_early_failure::case_3_missing_runtime_digest ... ok
test se::tests::test_get_evidence_early_failure::case_4_digest_too_large ... ok
test se::tests::test_request_all_fields_survive_roundtrip ... ok
test se::tests::test_request_digest_none_omitted_from_json ... ok
test se::tests::test_request_digest_some_survives_roundtrip ... ok
test se::tests::test_response_all_fields_survive_roundtrip ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_1_measurement ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_2_additional_data ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_4_cuid ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_3_user_data ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_5_encr_measurement_key ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_6_encr_request_nonce ... ok
test se::tests::test_response_binary_fields_are_base64_strings::case_7_image_hdr_tags ... ok
test se::tests::test_se_report_data_size_is_sha512_length ... ok
test se::tests::test_se_attester_default_construction ... ok
test se::tests::test_user_data_to_bytes::case_1_empty ... ok
test se::tests::test_user_data_to_bytes::case_2_single_byte ... ok
test se::tests::test_user_data_to_bytes::case_3_all_bytes ... ok
test se::tests::test_user_data_to_bytes::case_4_production_digest ... ok
test se::tests::test_validate_and_pad_data_err::case_1_any_byte_into_zero_size ... ok
test se::tests::test_validate_and_pad_data_err::case_2_boundary_plus_one ... ok
test se::tests::test_validate_and_pad_data_err::case_3_well_above_limit ... ok
test se::tests::test_validate_and_pad_data_ok::case_1_zero_target_empty_input ... ok
test se::tests::test_validate_and_pad_data_ok::case_2_empty_padded_to_64 ... ok
test se::tests::test_validate_and_pad_data_ok::case_3_one_byte_padded ... ok
test se::tests::test_validate_and_pad_data_ok::case_4_four_bytes_padded ... ok
test se::tests::test_validate_and_pad_data_ok::case_5_sha256_size_padded ... ok
test se::tests::test_validate_and_pad_data_ok::case_6_boundary_minus_one ... ok
test se::tests::test_validate_and_pad_data_ok::case_7_exact_match ... ok
test se::tests::test_validate_and_pad_data_ok::case_8_all_byte_values_survive ... ok

test result: ok. 32 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s
```

Assisted-by: IBM Bob

### CODE REVIEW

File: attestation-agent/attester/src/se/mod.rs [hunk 1]
```diff
@@ -149,161 +149,323 @@ impl Attester for SeAttester {
...
+    // Fixtures — shared test data built once and injected by rstest
+    // ---------------------------------------------------------------------------
 
-    #[tokio::test]
-    async fn test_runtime_data_digest_serialization() {
-        // Test that runtime_data_digest is properly serialized and deserialized
-        let expected_digest = MOCK_RUNTIME_DIGEST.to_vec();
+    /// Zero-field BootHdrTags used wherever a real boot header is not needed.
+    #[fixture]
+    fn dummy_tags() -> BootHdrTags {
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs:165]: Where is this fixture used?

File: attestation-agent/attester/src/se/mod.rs [hunk 2]
```diff
@@ -149,161 +149,323 @@ impl Attester for SeAttester {
...
             measurement_size: 64,
             additional_size: 32,
-            encr_measurement_key: vec![0; 32],
-            encr_request_nonce: vec![0; 16],
-            image_hdr_tags: create_dummy_boot_hdr_tags(),
-            runtime_data_digest: Some(expected_digest.clone()),
-        };
+            encr_measurement_key: vec![0u8; 32],
+            encr_request_nonce: vec![0u8; 16],
+            image_hdr_tags: BootHdrTags::new([0u8; 64], [0u8; 64], [0u8; 64], [0u8; 16]),
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs:177]: Is it supposed to be used here?

File: attestation-agent/attester/src/se/mod.rs [hunk 3]
```diff
@@ -149,161 +149,323 @@ impl Attester for SeAttester {
...
+    //
+    // Note: boundary_plus_one and 65_bytes are kept as distinct named cases
+    // because one uses the SE_REPORT_DATA_SIZE constant and the other a literal,
+    // confirming both the constant's value and the literal boundary together.
+    // ---------------------------------------------------------------------------
+
+    #[rstest]
+    #[case(vec![0x01],                        0)] // any byte into 0-size
+    #[case(vec![0x42; SE_REPORT_DATA_SIZE+1], SE_REPORT_DATA_SIZE)] // boundary + 1 via const
+    #[case(vec![0xFF; 65],                    SE_REPORT_DATA_SIZE)] // boundary + 1 via literal
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs]: Does it the fact that one is a literal and one is a const make any difference to the test? The is another test for ` assert_eq!(SE_REPORT_DATA_SIZE, 64);`, so I'm not sure what the 65 literal adds here?

File: attestation-agent/attester/src/se/mod.rs [hunk 4]
```diff
@@ -149,161 +149,323 @@ impl Attester for SeAttester {
...
+    // ---------------------------------------------------------------------------
+    // UserData::to_bytes
+    // Contract: returns the exact stored bytes, nothing added or removed.
+    // ---------------------------------------------------------------------------
+
+    #[rstest]
+    #[case(vec![])] // empty — identity on empty slice
+    #[case(vec![0x42])] // single byte
+    #[case((0u8..=255).collect())] // all 256 byte values, order preserved
+    #[case(MOCK_RUNTIME_DIGEST.to_vec())] // production-sized digest
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs]: Using case descriptions rather than comments is better:

```suggestion
    #[case::empty(b"")]
    #[case::single_byte(&[0x42])]
    #[case::all_bytes(&(0..=255).collect::<Vec<u8>>())]
    #[case::production_digest(MOCK_RUNTIME_DIGEST)]
```

File: attestation-agent/attester/src/se/mod.rs [hunk 5]
```diff
@@ -149,161 +149,323 @@ impl Attester for SeAttester {
...
+    fn test_user_data_to_bytes(#[case] digest: Vec<u8>) {
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs]: I think that using `&u8` for parameters rather than Vec is more idiomatic as it's more flexible and saves a memory allocate/clone under certain circumstances, like here where you will do the heap allocation on each case followed by a clone in the test body.
```suggestion
    fn test_user_data_to_bytes(#[case] digest: &[u8]) {
```

File: attestation-agent/attester/src/se/mod.rs [hunk 6]
```diff
@@ -149,161 +149,323 @@ impl Attester for SeAttester {
...
-            assert_eq!(result.len(), SE_REPORT_DATA_SIZE);
-
-            // Verify original data is preserved at the beginning
-            assert_eq!(&result[..original_len], &small_digest[..]);
-
-            // Verify zeros are padded at the end
-            for i in original_len..SE_REPORT_DATA_SIZE {
-                assert_eq!(result[i], 0, "Byte at index {} should be 0 (padded)", i);
+    #[test]
+    fn test_user_data_clone_is_independent() {
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs]: This is just testing the `Vec<u8>::clone` logic? I'm not sure we need to test rust library features?

**Review by stevenhorsman — COMMENTED** [review]
Some comments/suggestions

File: attestation-agent/attester/src/se/mod.rs (hunk 6 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs]: Yeah. Thanks for ponting it. i have removed these tescases now.

File: attestation-agent/attester/src/se/mod.rs (hunk 5 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs]: Yes i agree. changed it. Thanks

File: attestation-agent/attester/src/se/mod.rs (hunk 4 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs]: Changed as per the suggestion. Thanks

File: attestation-agent/attester/src/se/mod.rs (hunk 3 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs]: Yes its not required. removed it Thanks for poitning it out.

File: attestation-agent/attester/src/se/mod.rs (hunk 2 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs:177]: Yes i have reused the fixture. Thanks

File: attestation-agent/attester/src/se/mod.rs (hunk 4 above)
Comment (stevenhorsman) [reply attestation-agent/attester/src/se/mod.rs]: I meant everywhere, not just this single test

Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs]: Yeah missed this. chagned all now. Please review. Thanks

File: attestation-agent/attester/src/se/mod.rs [hunk 7]
```diff
@@ -149,161 +149,277 @@ impl Attester for SeAttester {
...
+        assert!(
+            !json.contains("runtime_data_digest"),
+            "key must be absent when None, got: {json}"
+        );
+        let parsed: SeAttestationRequest = serde_json::from_str(&json).unwrap();
+        assert!(parsed.runtime_data_digest.is_none());
+    }
 
+    #[rstest]
+    fn test_request_all_fields_survive_roundtrip(dummy_tags: BootHdrTags) {
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs:326]: This doesn't assert image_hdr_tags round-trips? Every other field on SeAttestationRequest is checked explicitly, but image_hdr_tags is silently omitted. If BootHdrTags ever loses its serde impl, this test won't catch it.

File: attestation-agent/attester/src/se/mod.rs [hunk 8]
```diff
@@ -149,161 +149,277 @@ impl Attester for SeAttester {
...
+    // SeAttestationResponse serialization
+    // ---------------------------------------------------------------------------
+
+    #[rstest]
+    fn test_response_all_fields_survive_roundtrip(dummy_response: SeAttestationResponse) {
+        // All Base64 fields must round-trip without corruption.
+        let parsed: SeAttestationResponse =
+            serde_json::from_str(&serde_json::to_string(&dummy_response).unwrap()).unwrap();
+
+        assert_eq!(parsed.measurement, dummy_response.measurement);
```
Comment (stevenhorsman) [inline attestation-agent/attester/src/se/mod.rs:362]: Similarly `cuid` and `image_hdr_tags` aren't checked here?

**Review by stevenhorsman — COMMENTED** [review]
A couple of things to check

File: attestation-agent/attester/src/se/mod.rs (hunk 8 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs:362]: Added the new logic to verify the missing properties. Thanks

File: attestation-agent/attester/src/se/mod.rs (hunk 7 above)
Comment (SantoshMadhukar-K) [reply attestation-agent/attester/src/se/mod.rs:326]: Added the new logic to verify the missing properties. Thanks for pointing it out

**Review by stevenhorsman — APPROVED** [review]
LGTM.

**Review by Xynnn007 — APPROVED** [review]
